packages/testlab/src/server-options.ts

```diff
@@ -0,0 +1,19 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/testlab
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+/*
+For @bajtos: We need the `ApplicationConfig` and `HttpServerOptions` shapes,
+but importing them respectively from `@loopback/core` and `@loopback/http-server`,
+respectively will create cyclic dependencies.
+
+What's the best way to resolve this?
+*/
+
+// export function givenServerOptions(
```

Comment:
`testlab` should contain utilities that will be used by most if not all of the other packages. For server stuff ... Options is probably best configured within the package that needs this -- in a helper method probably within that package not in testlab and it'll help avoid the cyclic dependency.

Reply:
Server options is a thing we need in multiple packages, namely `http-server` and `rest`. That's why I proposed to move this helper to testlab.

Looking at `givenServerOptions` in `rest.integration.test.ts`, I think we use a wrong type signature there. Application config contains the REST/HTTP configuration nested under `rest` property, but the helper is returning options at the root level.

IMO, `testlab` should provide a helper for building http-server options only. To avoid cyclic dependencies, I am proposing the following implementation in `testlab`:

```ts
function givenHttpServerConfig<T>(options?: T): T & {host?: string} {
  const defaults = process.env.TRAVIS ? {host: '127.0.0.1'} : {};
  return Object.assign(defaults, options);
}
```